projects/ngx-vflow-lib/src/lib/vflow/math/edge-path/bezier-path.ts

```diff
@@ -12,175 +12,56 @@ export function bezierPath(
-    return bezierPathBtt(source, target, usingPoints)
-  }
 
-  throw new Error('Unhandled combination of sourcePosition and targetPosition')
-}
-
-/**
- * Left-to-right direction
- */
-function bezierPathLtr(source: Point, target: Point, usingPoints: UsingPoints): PathData {
   const path = d3Path()
 
   path.moveTo(source.x, source.y)
 
   let firstControl: Point
   let secondControl: Point
 
-  if (source.x > target.x) {
-    const distance = source.x - target.x
-
-    // TODO: probably need to make this configurable
-    const curvature = .25
-    // thanks colleagues from react/svelte world
-    // https://github.com/xyflow/xyflow/blob/f0117939bae934447fa7f232081f937169ee23b5/packages/system/src/utils/edges/bezier-edge.ts#L56
-    const controlOffset = curvature * 25 * Math.sqrt(distance)
-
-    firstControl = { x: source.x + controlOffset, y: source.y }
-    secondControl = { x: target.x - controlOffset, y: target.y }
-  } else {
-    const middleX = (source.x + target.x) / 2
-
-    firstControl = { x: middleX, y: source.y }
-    secondControl = { x: middleX, y: target.y }
-  }
-
-  path.bezierCurveTo(
-    firstControl.x, firstControl.y,
-    secondControl.x, secondControl.y,
-    target.x, target.y
-  )
-
-  return getPathData(path, source, target, firstControl, secondControl, usingPoints)
-}
-
-/**
- * Right-to-left direction
- */
-function bezierPathRtl(source: Point, target: Point, usingPoints: UsingPoints): PathData {
-  const path = d3Path()
-
-  path.moveTo(source.x, source.y)
-
-  let firstControl: Point
-  let secondControl: Point
-
-  if (source.x < target.x) {
-    const distance = target.x - source.x
-
-    // TODO: probably need to make this configurable
-    const curvature = .25
-    // thanks colleagues from react/svelte world
-    // https://github.com/xyflow/xyflow/blob/f0117939bae934447fa7f232081f937169ee23b5/packages/system/src/utils/edges/bezier-edge.ts#L56
-    const controlOffset = curvature * 25 * Math.sqrt(distance)
-
-    firstControl = { x: source.x - controlOffset, y: source.y }
-    secondControl = { x: target.x + controlOffset, y: target.y }
-  } else {
-    const middleX = (source.x + target.x) / 2
-
-    firstControl = { x: middleX, y: source.y }
-    secondControl = { x: middleX, y: target.y }
-  }
-
-  path.bezierCurveTo(
-    firstControl.x, firstControl.y,
-    secondControl.x, secondControl.y,
-    target.x, target.y
-  )
-
-  return getPathData(path, source, target, firstControl, secondControl, usingPoints)
-}
-
-/**
- * Bottom-to-top
- */
-function bezierPathBtt(source: Point, target: Point, usingPoints: UsingPoints): PathData {
-  const path = d3Path()
-
-  path.moveTo(source.x, source.y)
-
-  let firstControl: Point
-  let secondControl: Point
-
-  if (source.y < target.y) {
-    const distance = target.y - source.y
+    firstControl = {x: source.x, y: source.y}
+    secondControl = {x: target.x, y: target.y}
+
+    function calcContrPoint (
```

Comment:
It's better to move this function outsize of bezierPath due to perf reasons.

Also you could add additional arguments to pass source and target points to match such signature:

```typescript
function calcControlPoint(
  point: Point,
  pointPosition: Position,
  source: Point,
  target: Point
)
```

then you could just call

```typescript
  const firstControl = calcControlPoint(source, sourcePosition, source, target)
  const secondControl = calcControlPoint(target, targetPosition, source, target)
```

Also should rename calcContrPoint -> calcControlPoint

Reply:
1. Renamed the function calcContrPoint -> calcControlPoint
2. Moved this function outsize of bezierPath
3. Add additional arguments "distance"